ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -449,9 +452,76 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 				return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, err
 			}
 		}
+		if rayJobInstance.Spec.Schedule != "" && rayJobInstance.Status.JobDeploymentStatus != rayv1.JobDeploymentStatusFailed {
+			logger.Info("RayJob is scheduled again")
+			rayJobInstance.Status.JobDeploymentStatus = rayv1.JobDeploymentStatusScheduling
+		} else {
+			// NOTE: we do not requeue if the job fails even if scheduled, this could change
+			// If the RayJob is completed without scheduling or has failed, we should not requeue it.
+			logger.Info("RayJob is not scheduled")
+			return ctrl.Result{}, nil
+		}
+	case rayv1.JobDeploymentStatusScheduling:
```

Comment:
Thinking about this switch case more, I'm wondering why we have deletion logic here in the first place. If a RayJob transitions to `Completed` or `Failed` state before `Scheduling`, it should be deleted in the `case rayv1.JobDeploymentStatusComplete, rayv1.JobDeploymentStatusFailed` block. Am I missing something?